test/setup/iam.tf

```diff
@@ -15,22 +15,79 @@
  */
 
 locals {
-  int_required_roles = [
-    "roles/bigquery.admin",
-    "roles/aiplatform.admin",
-    "roles/cloudfunctions.admin",
-    "roles/dataform.admin",
-    "roles/datalineage.viewer",
-    "roles/iam.serviceAccountAdmin",
-    "roles/iam.serviceAccountTokenCreator",
-    "roles/iam.serviceAccountUser",
-    "roles/logging.configWriter",
-    "roles/resourcemanager.projectIamAdmin",
-    "roles/run.invoker",
-    "roles/serviceusage.serviceUsageAdmin",
-    "roles/storage.admin",
-    "roles/workflows.admin"
-  ]
+  per_module_roles = {
+    root = [
+      "roles/bigquery.admin",
+      "roles/iam.serviceAccountAdmin",
```

Comment:
* Is`roles/iam.serviceAccountAdmin` and `roles/resourcemanager.projectIamAdmin` roles needed? which resource requires this?

* Should below roles be included?
`roles/storage.objectViewer`, for creating external table based on GCS bucket.
`roles/cloudkms.cryptoKeyEncrypterDecrypter` : if kms key is provided for default_encryption_configuration in dataset.